src/prefect/cli/register.py

```diff
@@ -74,15 +80,19 @@ def flow(file, name, project, label):
         --project, -p   TEXT    The name of a Prefect project to register this flow
         --label, -l     TEXT    A label to set on the flow, extending any existing labels.
                                 Multiple labels are supported, eg. `-l label1 -l label2`.
+        --detect-changes        If toggled, passes a serialized hash of the flow to the register
+                                function's idempotency key. This allows you to avoid bumping the
+                                registered flow's version if the flow has not changed.
```

Comment:
We have to be really careful here because the serialized hash is **not** a great representation of whether or not the flow itself has changed.

Reply:
Agreed. Perhaps this should be renamed to something more verbose/descriptive of what it does. Perhaps

```
prefect register flow --skip-if-flow-structure-unchanged
```

Reply:
Right right, we weren't exactly sure the best way to phrase this / what would make it more clear. It is specifically for the case where the metadata has not changed. Open for suggestions!

Reply:
Metadata may be a good word. Perhaps

```
prefect register flow --skip-if-flow-metadata-unchanged
```

Reply:
`--only-on-metadata-changes` ? 🤷 tricky stuff

Reply:
I like `skip` as the prefix, since we use that elsewhere already. No thoughts on the suffix.

Reply:
What if we used `--idempotency-key` (which is very explicit), and if no special key is provided we default to `flow.serialized_hash()`?  That way we match the register kwarg without misleading anyone.

Reply:
`--idempotency-key` sounds like it takes a value, so it'd be weird to have it optionally not take a value. It's also a bit uncommon for a CLI flag to take 0 or 1 args, so this may confuse users. I can't think of a reason why a user would pass in an explicit idempotency key that wasn't the flow hash, and it's also not clear how a user would write a bash script that made use of a flow hash (how would they hash the flow from the CLI? hardcoding the value feels wrong).

I'd prefer `--skip-if-flow-metadata-unchanged` (or something like it), since this is what most users would want and is explicit about the behavior. If you're concerned about api compatibility, we might add an explicit flag to `flow.register` as well with the same name, so the apis exist in both python code and CLI.